### ISSUE TITLE
Vanilla UNet - Model card

### ISSUE THREAD

**Issue by punithsekar** [tenstorrent/tt-metal#13272]
Reference link:- https://github.com/mateuszbuda/brain-segmentation-pytorch/tree/d45f8908ab2f0246ba204c702a6161c9eb25f902

Vanilla Unet torch implementation of our reference implementation  - https://github.com/tenstorrent/tt-metal/blob/05a8838c0d935d41112e07bc867b4cfc40115d09/models/experimental/functional_vanilla_unet/reference/unet.py

Our pytorch model graph - 
[model_vanilla_unet.gv.pdf](https://github.com/user-attachments/files/17187182/model_vanilla_unet.gv.pdf)

Modules: 
1. UNet

Ops: 
1. Conv2d
2. BatchNorm2d
3. ReLU
4. MaxPool2d
5. cat
6. ConvTranspose2d
7. Sigmoid

**Comment by punithsekar** [comment]
fyi @saichandax

**Comment by punithsekar** [comment]
Closing this for now, We can open it if needed!

**Comment by saichandax** [comment]
Model is in main. https://github.com/tenstorrent/tt-metal/tree/main/models/demos/vanilla_unet
